maintenance/wikia/Forum/fixCommentIndexes.php

```diff
@@ -0,0 +1,262 @@
+	 * un-deleting/removing/archiving as a thing.  Not going to fix that here since who knows what
+	 * additional bugs that will uncover.
+	 *
+	 * @return array
+	 */
+	private function getAffectedThreads() {
+		$dbr = wfGetDB( DB_SLAVE );
+		$threads = ( new WikiaSQL )
+			->SELECT( 'parent_comment_id', 'props', 'count(*)' )
+			->FROM( 'comments_index' )
+			->JOIN( 'page_wikia_props' )
+			->WHERE( 'page_id' )->EQUAL_TO_FIELD( 'comment_id' )
+			->AND_( 'propname' )->EQUAL_TO( WPP_WALL_COUNT )
+			->AND_( 'parent_comment_id' )->NOT_EQUAL_TO( 0 )
+			->AND_( 'removed' )->EQUAL_TO( 0 )
+			->AND_( 'deleted' )->EQUAL_TO( 0 )
+			->AND_( 'archived' )->EQUAL_TO( 0 )
+			->GROUP_BY( 'parent_comment_id, props' )
+			->HAVING( 'count(*)' )->GREATER_THAN( 1 )
+			->runLoop( $dbr, function ( &$threads, $row ) {
+				$threads[] = $row->parent_comment_id;
+			} );
+
+		// Ensure that we're always sending an array back
+		return empty( $threads ) ? [ ] : $threads;
+	}
+
+	/**
+	 * Get the URL for the wiki being worked on by this script
+	 *
+	 * @return string
+	 */
+	static public function getWikiURL() {
+		$dbName = WikiFactory::IDtoDB( F::app()->wg->CityId );
+		$url = WikiFactory::DBtoDomain( $dbName );
+		return WikiFactory::getLocalEnvURL( 'http://'.$url );
+	}
+}
+
+class BrokenThread {
+
+	private $threadId;
+
+	/**
+	 * @param int $threadId
+	 */
+	public function __construct( $threadId ) {
+		$this->threadId = $threadId;
+	}
+
+	/**
+	 * Fix the comment numbering for the thread given by $this->threadId
+	 */
+	public function fix() {
+		FixCommentIndexes::debug( "* Fixing thread ".$this->threadId );
+
+		$lastIdx = $this->renumberComments();
+		$this->setNewWallCount( $lastIdx );
+	}
+
+	/**
+	 * Sets the new index to use whenever the next comment is made on the current thread
+	 *
+	 * @param int $idx Update the new last used index on this wall
+	 */
+	private function setNewWallCount( $idx ) {
+		if ( FixCommentIndexes::isTest() ) {
+			return;
+		}
+
+		wfSetWikiaPageProp( WPP_WALL_COUNT, $this->threadId, $idx );
+	}
+
+	/**
+	 * Update the comment number used for creating links to those comments.  It doesn't matter what they are
+	 * as long as each comment has a unique number, however ascending from 2 is how it works now and makes the
+	 * most sense.
+	 *
+	 * @return int The last index we assigned to a comment
+	 */
+	private function renumberComments() {
+		$comments = $this->getCommentIds();
+
+		// Output some info if we're verbose.
+		FixCommentIndexes::debug(" -- ".$this->getThreadURL()."\n" );
+		FixCommentIndexes::debug( "-- Found ".count($comments)." comments\n" );
+		FixCommentIndexes::debug( "-- Renumbering ... " );
+
+		$commentIdx = 0;
+		$lastIdx = 0;
+		$foundDuplicates = false;
+
+		foreach ( $comments as $commentId ) {
+			$thisIdx = wfGetWikiaPageProp( WPP_WALL_COUNT, $commentId );
+			$isDuplicate = $thisIdx == $lastIdx;
+
+			// Run this when we find the first duplicate
+			if ( $isDuplicate && !$foundDuplicates ) {
```

Comment:
Probably those conditions could be simplified to one flag:
```$foundDuplicates |= $thisIdx == $lastIdx```

Then just using preincrementation for updates.